xdsl/irdl.py

```diff
@@ -782,12 +983,17 @@ def wrong_field_exception(field_name: str) -> PyRDLOpDefinitionError:
             # IRDLOperation ClassVar fields are allowed
             if field_name in ["irdl_options", "traits", "name"]:
                 continue
+
+            if field_name in operation_fields:
+                continue
+
             # Dunder fields are allowed (i.e. __orig_bases__, __annotations__, ...)
             # They are used by Python to store information about the class, so they
             # should not be considered as part of the operation definition.
             # Also, they can provide a possiblea escape hatch.
-            if field_name[:2] == "__" and field_name[-2:] == "__":
+            if field_name.startswith("__") and field_name.endswith("__"):
```

Comment:
Minor change, I think this is the recommended way to do this on the offchance the field name is a single character

Reply:
- [ ] I'll remove it